aya/src/maps/xdp/dev_map.rs

```diff
@@ -0,0 +1,179 @@
+//! An array of network devices.
+
+use std::{
+    borrow::{Borrow, BorrowMut},
+    num::NonZeroU32,
+    os::fd::{AsFd, AsRawFd},
+};
+
+use aya_obj::generated::{bpf_devmap_val, bpf_devmap_val__bindgen_ty_1};
+
+use crate::{
+    maps::{check_bounds, check_kv_size, IterableMap, MapData, MapError},
+    programs::ProgramFd,
+    sys::{bpf_map_lookup_elem, bpf_map_update_elem, SyscallError},
+    Pod, FEATURES,
+};
+
+/// An array of network devices.
+///
+/// XDP programs can use this map to redirect to other network
+/// devices.
+///
+/// # Minimum kernel version
+///
+/// The minimum kernel version required to use this feature is 4.14.
+///
+/// # Examples
+/// ```no_run
+/// # let mut bpf = aya::Bpf::load(&[])?;
+/// use aya::maps::xdp::DevMap;
+///
+/// let mut devmap = DevMap::try_from(bpf.map_mut("IFACES").unwrap())?;
+/// let source = 32u32;
```

Comment:
We should probably use more realistic indexes :P

Reply:
Indexes do grow pretty fast as soon as you have vms or containers involved. But sure. Changed to 2 and 3, and added a comment not to think the key/index is an interface index (though it's a common usecase, but it does not have to be)